gitpod-io/monitoring-cell
# Refactor status API

## Description

This PR focuses on 2 changes (split by commit)

1. Refactor the reconcile loop, making it independent from the Status API. Now the Status is updated only at the end of the loop, and the reconciliation is dependent on the real world instead of the status
2. The Status API was refactored to provide useful information instead of a simplistic boolean

## Related Issue(s)

Fixes #19 

## How to test

1. Follow instructions from README to deploy the operator
2. Run `watch kubectl get cell cell-sample -o yaml` and see the status changing over time
